truss/remote/baseten/api.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from typing import Optional
+
+import requests
+from truss.remote.baseten.auth import AuthService
+from truss.remote.baseten.error import ApiError
+
+logger = logging.getLogger(__name__)
+
+
+class BasetenApi:
+    def __init__(self, api_url: str, auth_service: Optional[AuthService] = None):
+        self._api_url = api_url
+        self._auth_service = auth_service or AuthService()
+        self._auth_token = self._auth_service.authenticate()
+
+    def _post_graphql_query(self, query_string: str) -> dict:
+        headers = self._auth_token.headers()
+        resp = requests.post(
+            self._api_url,
+            data={"query": query_string},
+            headers=headers,
+        )
+
+        if not resp.ok:
+            logger.error(f"GraphQL endpoint failed with error: {resp.content}")  # type: ignore
+            resp.raise_for_status()
```

Comment:
curios, why resp.ok first and not raise_for_status always and log the error in except clause.